instrumentation/servlet/servlet-3.0/javaagent/build.gradle.kts

```diff
@@ -37,9 +37,18 @@ dependencies {
   latestDepTestLibrary("org.apache.tomcat.embed:tomcat-embed-jasper:9.+") // see servlet-5.0 module
 }
 
-tasks.withType<Test>().configureEach {
-  jvmArgs("-Dotel.instrumentation.servlet.experimental.capture-request-parameters=test-parameter")
-  // required on jdk17
-  jvmArgs("--add-opens=java.base/java.util=ALL-UNNAMED")
-  jvmArgs("-XX:+IgnoreUnrecognizedVMOptions")
+tasks {
+  withType<Test>().configureEach {
+    jvmArgs("-Dotel.instrumentation.servlet.experimental.capture-request-parameters=test-parameter")
+    // required on jdk17
+    jvmArgs("--add-opens=java.base/java.util=ALL-UNNAMED")
+    jvmArgs("-XX:+IgnoreUnrecognizedVMOptions")
+  }
+  val testSnippetInjection by registering(Test::class) {
+    jvmArgs("-Dotel.experimental.javascript-snippet=<script> Test </script>")
```

Comment:
Running all tests twice, once without snippet injection and then with does not seem reasonable. Have you considered moving snippet tests to a separate class so they could be run independently or running the tests always with snippet injection enabled.